config/plugins/sonic-pbh_yang.py

```diff
@@ -0,0 +1,747 @@
+        if value is None and attr in cfg[table][key]:
+            cfg[table][key].pop(attr)
+        else:
+            cfg[table][key][attr] = value
+
+    db.set_entry(table, key, cfg[table][key])
+
+
+def del_entry(db, table, key):
+    """ Delete entry in table """
+
+    cfg = db.get_config()
+    cfg.setdefault(table, {})
+    if key not in cfg[table]:
+        raise Exception(f"{key} does not exist")
+
+    cfg[table].pop(key)
+
+    db.set_entry(table, key, None)
+
+
+def ip_address_validator(ctx, param, value):
+    """ Check if the given ip address is valid """
+
+    if value is not None:
+        try:
+            ip = ipaddress.ip_address(value)
+        except Exception as e:
+            exit_with_error("Error: invalid value for '{}' option\n{}".format(param.name, e), fg="red")
+
+        return str(ip)
+
+
+def pbh_match(ctx, param, value):
+    """ Check if PBH rule options is valid """
+
+    if value is not None:
+        if re.match(pbh_pattern, str(value)) is None:
+            exit_with_error("Error: invalid value for '{}' option".format(param.name), fg="red")
+
+        return value
+
+
+def is_exist_in_db(db, _list, conf_db_key, option_name):
+    """ Check if provided CLI option already exist in Config DB,
+        i.g in case of --hash-field-list option it will check
+        if 'hash-field' was previously added by
+        'config pbh hash-field ...' CLI command
+
+        Args:
+            db: reference to Config DB,
+            _list: value of 'click' option
+            conf_db_key: key to search in Config DB
+            option_name: name of 'click' option
+    """
+
+    if _list is None:
+        return
+
+    table = db.cfgdb.get_table(conf_db_key)
+    correct_list = list(table.keys())
+
+    splited_list = _list.split(',')
+
+    for elem in splited_list:
+        if elem not in correct_list:
+            exit_with_error("Error: invalid value '{}' for '{}' option, please use {}".format(elem, option_name, correct_list), fg="red")
+
+
+def ip_mask_hash_field_correspondence(ip_mask, hash_field):
+    # at this point
+    # --ip_mask value can be None or VALID IP
+    # --hash_field always valid
+
+    if ((ip_mask is None) and
+        (hash_field == 'INNER_DST_IPV4' or hash_field == 'INNER_SRC_IPV4' or
+        hash_field == 'INNER_DST_IPV6' or hash_field == 'INNER_SRC_IPV6')):
+        exit_with_error("Error: the value of --hash-field='{}' is NOT compatible with the value of --ip-mask='{}'".format(hash_field, ip_mask), fg='red')
+
+    if ((ip_mask is not None) and
+        (hash_field == 'INNER_IP_PROTOCOL' or
+        hash_field == 'INNER_L4_DST_PORT' or
+        hash_field == 'INNER_L4_SRC_PORT')):
+        exit_with_error("Error: the value of --hash-field='{}' is NOT compatible with the value of --ip-mask='{}'".format(hash_field, ip_mask), fg='red')
+
+    if ((hash_field == 'INNER_DST_IPV4' or hash_field == 'INNER_SRC_IPV4') and
+        (not isinstance(ipaddress.ip_address(ip_mask), IPv4Address))):
```

Comment:
@vadymhlushko-mlnx please optimize IP mask check: do it once if `ip_mask` is not `None`:
```python
>>> import ipaddress
>>> ipaddress.ip_address(u'255.0.0.0').version
4
>>> ipaddress.ip_address(u'ffff::').version
6
```